Composer: fix incompatible dependencies

jrfnl opened the pull request:
## Description

The `composer.lock` file has locked in a number of dependencies at version which are not compatible with the lowest PHP version the plugin is supposed to be compatible with.

By setting `config -> platform` in the `composer.json` file, the PHP version referenced there will be used for the version negotiations, preventing these type of incompatibilities from happening in the future.

## Testing Instructions

Using the `develop` branch:
- Set the PHP version used on the system to PHP 5.6 and try running `composer install`. It will fail.

Using this branch:
- Set the PHP version used on the system to PHP 5.6 and try running `composer install`. It will succeed.
- Set the PHP version used on the system to a higher PHP version and see it succeed again.

JasonTheAdams commented:
@jrfnl Thanks for the PR! I apologize it took so long to get to this, to be honest I completely missed it.

This is a good suggestion, though! I'll resolve the merge conflict and get this merged in time for the next release.

Thank you!

jrfnl commented:
@JasonTheAdams No worries, 24 days is nothing compared to some repos ;-)

As `composer.lock` conflicts are horrible to solve, I've just rebased the PR and fixed it up myself. The PR should be mergable again now.

JasonTheAdams commented:
@jrfnl I'm having a strange Github issue. I pulled your PR and added a line to the changelog. I see the [change](https://github.com/jrfnl/givewp/commit/3a1ce778ab81156f2dfc50393310bda3258a93eb) on your branch, but the commit isn't showing up in the PR. Any ideas?

jrfnl commented:
@JasonTheAdams Might have been a browser cache issue or something. It is showing for me, at least, it is now.

JasonTheAdams commented:
Oh browser caching — my friend _and_ foe. 😆

JasonTheAdams commented:
Much appreciated, @jrfnl!
